crates/derive/src/batch/span_batch/batch.rs

```diff
@@ -57,6 +62,89 @@ impl SpanBatch {
         self.parent_check == hash[..20]
     }
 
+    /// Perform holocene checks on the span batch.
+    pub async fn is_batch_holocene_valid<BF: L2ChainProvider>(
+        &self,
+        l2_safe_head: L2BlockInfo,
+        block_time: u64,
+        fetcher: &mut BF,
+    ) -> bool {
```

Comment:
Can we use `BatchValidity`? There are circumstances within these checks where we're undecided, not `Drop`.